File: server/models/Tournament.ts
import mongoose, { Schema } from 'mongoose';
import { INewTournamentModel } from '../../constants/interfaces';

const Tournament: Schema = new Schema({
  adminUser: { type: String, required: true },
  hostClub: { type: String, required: true },
  tournamentName: { type: String, required: true },
  events: { type: Array, required: true },
  startDate: { type: Date, required: true },
  endDate: { type: Date, required: true },
  registrationCost: { type: Number, required: true },
  uuid: { type: String, required: true, unique: true },
});

export default mongoose.model<INewTournamentModel>('tournaments', Tournament);

File: HemaSite/Models/Tournament.cs
using System;
using System.Collections.Generic;

namespace HemaSite.Models
{
  public class Tournament
  {
    public int Id { get; set; }
    public string AdminUser { get; set; }
    public string HostClub { get; set; }
    public string TournamentName { get; set; }
    public ICollection<Event> Events { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public double RegistrationCost { get; set; }
    public ICollection<Competitor> Competitors { get; set; }
    public ICollection<Match> Matches { get; set; }
  }
}